main.tf

```diff
@@ -134,7 +134,7 @@ locals {
 data "aws_route53_zone" "this" {
   count = local.create_domain_name && var.create_domain_records ? 1 : 0
 
-  name = local.stripped_domain_name
+  name = coalesce(var.hosted_zone_name, local.stripped_domain_name)
```

Comment:
This doesnt allow null in both parameters.. breaks the previous API

Reply:
I don't think that's accurate. 
1.) A lookup without name doesn't make sense and would fail anyway
2.) `local.stripped_domain_name` cannot be null since we would fail a couple lines up already when calling `replace`